File: src/main/java/com/zaid/patel/AI_Email_Assistant/Controller/AuthController.java
package com.zaid.patel.AI_Email_Assistant.Controller;

import com.fasterxml.jackson.databind.JsonNode;
import com.zaid.patel.AI_Email_Assistant.DTo.AuthResponse;
import com.zaid.patel.AI_Email_Assistant.DTo.LoginRequest;
import com.zaid.patel.AI_Email_Assistant.DTo.SignUpRequest;
import com.zaid.patel.AI_Email_Assistant.Service.SupabaseAuthService;
import org.springframework.http.*;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/auth")
@CrossOrigin(origins = "*")
public class AuthController {

    private final SupabaseAuthService authService;

    public AuthController(SupabaseAuthService authService) {
        this.authService = authService;
    }

    @PostMapping("/signup")
    public ResponseEntity<?> signUp(@RequestBody SignUpRequest request) {
        try {
            AuthResponse response = authService.signUp(request);
            return ResponseEntity.status(HttpStatus.CREATED).body(response);
        } catch (RuntimeException e) {
            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
                    .body(java.util.Map.of("error", e.getMessage()));
        }
    }

    @PostMapping("/login")
    public ResponseEntity<?> signIn(@RequestBody LoginRequest request) {
        try {
            AuthResponse response = authService.signIn(request);
            return ResponseEntity.ok(response);
        } catch (RuntimeException e) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                    .body(java.util.Map.of("error", e.getMessage()));
        }
    }

    @GetMapping("/profile")
    public ResponseEntity<?> getProfile(@RequestHeader("Authorization") String authHeader) {
        try {
            String token = authHeader.replace("Bearer ", "");
            JsonNode user = authService.getUser(token);
            return ResponseEntity.ok(user);
        } catch (RuntimeException e) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                    .body(java.util.Map.of("error", e.getMessage()));
        }
    }
}

File: apps/api/src/controllers/auth.controller.js
import * as authService from '../services/supabase-auth.service.js'

export async function signup(req, res, next) {
  try {
    const { email, password, name } = req.validatedBody
    const result = await authService.signup(email, password, name)
    res.status(201).json(result)
  } catch (err) {
    res.status(400).json({ error: err.message })
  }
}

export async function login(req, res, next) {
  try {
    const { email, password } = req.validatedBody
    const result = await authService.login(email, password)
    res.json(result)
  } catch (err) {
    res.status(401).json({ error: err.message })
  }
}

export async function profile(req, res, next) {
  try {
    const user = await authService.getProfile(req.accessToken)
    res.json(user)
  } catch (err) {
    res.status(401).json({ error: err.message })
  }
}
